### PR TITLE
Updating the msbuild templates in the CLI to use TargetFramework

### PR DESCRIPTION
**Pull request by livarcocc**
Updating the msbuild templates in the CLI to use TargetFramework (singular).

@eerhardt @brthor @piotrpMSFT @jgoshi

### CODE REVIEW

**Comment by brthor** [comment]
migration changes look good

**Comment by livarcocc** [comment]
@dotnet-bot Test CentOS7.1 x64 Debug Build

**Comment by eerhardt** [comment]
:shipit:

File: src/dotnet/commands/dotnet-restore3/Program.cs [hunk 1]
```diff
@@ -118,5 +120,10 @@ public static int Run(string[] args)
 
             return cmd.Execute(args);
         }
+
+        private static void AddWorkaroundForOutputValidationIssue(IList<string> msbuildArgs)
+        {
+            msbuildArgs.Add("/p:SkipInvalidConfigurations=true;_InvalidConfigurationWarning=false");
```
Comment (rainersigwald) [inline src/dotnet/commands/dotnet-restore3/Program.cs]: This could be in its own commit for easier removal, but not a big deal.

**Comment by livarcocc** [comment]
@dotnet-bot Test RHEL7.2 x64 Release Build

**Comment by livarcocc** [comment]
@dotnet-bot Test Ubuntu x64 Release Build

**Comment by livarcocc** [comment]
@dotnet-bot Test Ubuntu x64 Release Build
